From f2b45e6a18a0624032d289318569ad57c24fd419 Mon Sep 17 00:00:00 2001
From: Jakob Unterwurzacher <jakobunt@gmail.com>
Date: Wed, 12 Feb 2020 22:19:49 +0100
Subject: [PATCH] earlyoom.service: drop root privileges

Run as a random unprivilege user instead of as root,
but add the capabilities CAP_KILL CAP_IPC_LOCK.

Supersedes https://github.com/rfjakob/earlyoom/pull/158
---
 earlyoom.service.in | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/earlyoom.service.in b/earlyoom.service.in
index 5f193f0..2df9330 100644
--- a/earlyoom.service.in
+++ b/earlyoom.service.in
@@ -5,6 +5,14 @@ Documentation=man:earlyoom(1) https://github.com/rfjakob/earlyoom
 [Service]
 EnvironmentFile=-:SYSCONFDIR:/default/earlyoom
 ExecStart=:TARGET:/earlyoom $EARLYOOM_ARGS
+# Run as an unprivileged user with random user id
+DynamicUser=true
+# Allow killing processes and calling mlockall()
+AmbientCapabilities=CAP_KILL CAP_IPC_LOCK
+# We don't need write access anywhere
+ProtectSystem=strict
+# We don't need /home at all, make it inaccessible
+ProtectHome=true
 
 [Install]
 WantedBy=multi-user.target
